features/step_definitions/journey_test_steps.rb

```diff
@@ -47,6 +68,10 @@
   click_link_or_button(button_name)
 end
 
+When /I click the '(.*)' button for the supplier user '(.*)'$/ do |button_name,user_name|
+  find(:xpath, "//*/span[contains(text(),'#{user_name}')]/../../td/*//button[contains(text(),'#{button_name}')]").click
```

Comment:
I'd be wary of adding things to the HTML which are for the tests. The problem with that is you're trying to make the tests more readable by adding stuff to the HTML that only makes sense if you're trying to read the tests. If I come to the HTML trying to understand something else about it, I'll be wondering why those things are there because they only make sense from the perspective of the tests.

In my opinion this is a limitation of our testing language only using XPath. Could we not break it up a bit more into helper functions so it better describes the bit you're testing as you would describe it yourself?

For example, I'd say it as: ' the activate button for the email row of the supplier table' which we could split into these helpers:

1. get supplier table (get table with caption of 'supplier')
2. get table row for email field (get row of table where 1st cell contains 'email' text)
3. get activate button for a table row (get button in last cell of table row)

get_table_called('supplier').get_row_for('email').get_button('activate')

?